### PR TITLE
EDU-863: DataConverter serialization

### PR DESCRIPTION
**Pull request by DJSanti**
## What does this PR do?
This PR adds a note about the default data conveter's inability to handle lists of Protobufs.

### CODE REVIEW

File: docs-src/concepts/what-is-a-default-data-converter.md [hunk 1]
```diff
@@ -26,3 +26,6 @@ For example:
 - If a value is an instance of a Protobuf message, it is encoded with [proto3 JSON](https://developers.google.com/protocol-buffers/docs/proto3#json).
 - If a value isn't null, binary, or a Protobuf, it is encoded as JSON.
   If any part of it is not serializable as JSON, <!--(for example, a Date—see JSON data types)--> an error is thrown.
+
+The default data converter cannot deserialize a list of Protobufs.
```
[the remark this answers is not in the document]
Comment (cretz) [reply docs-src/concepts/what-is-a-default-data-converter.md:30]: This is marked resolved, but there is no reply or doc change

[the remark this answers is not in the document]
Comment (DJSanti) [reply docs-src/concepts/what-is-a-default-data-converter.md]: Made the change. Had to push again. Please look again.

[the remark this answers is not in the document]
Comment (cretz) [reply docs-src/concepts/what-is-a-default-data-converter.md]: Looks good. I think the example about protobuf lists is nice, but not required.

**Review by cretz — APPROVED** [review]

File: docs-src/concepts/what-is-a-default-data-converter.md [hunk 2]
```diff
@@ -26,3 +26,6 @@ For example:
 - If a value is an instance of a Protobuf message, it is encoded with [proto3 JSON](https://developers.google.com/protocol-buffers/docs/proto3#json).
 - If a value isn't null, binary, or a Protobuf, it is encoded as JSON.
   If any part of it is not serializable as JSON, <!--(for example, a Date—see JSON data types)--> an error is thrown.
+
+The default Data Converter serializes objects based on their root type, rather than nested types.
+Some SDKs' JSON serializers cannot process lists with Protobuf children objects without implementing a [custom Data Converter](/concepts/what-is-a-custom-data-converter).
```
Comment (djmagee) [inline docs-src/concepts/what-is-a-default-data-converter.md:31]:
```suggestion
The JSON serializers of some SDKs cannot process lists with Protobuf children objects without implementing a [custom Data Converter](/concepts/what-is-a-custom-data-converter).
```

**Review by djmagee — APPROVED** [review]
Just one tweak to avoid an awkward possessive noun.

**Pull request merged**